conans/client/generators/cmake_common.py

```diff
@@ -234,6 +234,30 @@ def generate_targets_section(dependencies):
     endif()
 endmacro()
 
+macro(conan_set_std)
+  if (CMAKE_VERSION VERSION_LESS "3.1")
+    if(CONAN_STD_CXX_FLAG)
```

Comment:
I don't fully get the CONAN_STD_CXX_FLAG and CONAN_CMAKE_CXX_STANDARD, I guess something Cmake-ish

Reply:
`CONAN_STD_CXX_FLAG`: Only used in case the CMake < 3.1, it is directly the flag to pass to the compiler, because CMake < 3.1 do no support it another way. Includes the `-std=XXXX`. Conan uses the compilers_info.py to translate to the correct flag depending on each compiler/version.

`CONAN_CMAKE_CXX_STANDARD`: Used > 3.1. Do not include the `-std=XXX` is directly the value of the standard, 98, 11, 14, 17. Internally CMake will translate it to the `-std=xxx` taking also into account the `CMAKE_CXX_EXTENSIONS` to know if it should like with "gnu++11" or just "c++11".